### PR TITLE
[core][node/worker] processing staking and normal transactions together

### PR DESCRIPTION
**Pull request by gupadhyaya**
* For block proposal, process pending PoolTransactions in the TransactionsByPriceAndNonce order (regardless of normal or staking)
* For block verification, pull and put normal and staking transactions into the same PoolTransactions list and apply and process in TransactionsByPriceAndNonce order

### CODE REVIEW

File: node/worker/worker.go [hunk 1]
```diff
@@ -86,15 +83,36 @@ func (w *Worker) CommitTransactions(
 			continue
 		}
 		// Start executing the transaction
-		w.current.state.Prepare(tx.Hash(), common.Hash{}, len(w.current.txs))
+		w.current.state.Prepare(tx.Hash(), common.Hash{}, dbIndex)
+		dbIndex = dbIndex + 1
```
Comment (rlan35) [inline node/worker/worker.go:87]: Here the index counter should only increment when the txn succeeded. Failed txns won't be included in the new block.

File: core/types/tx_pool.go [hunk 2]
```diff
@@ -102,3 +103,97 @@ type PoolTxByNonce PoolTransactions
...
+	heads  PoolTxByPrice                       // Next transaction for each unique account (price heap)
+	signer Signer                              // Signer for the set of transactions
+}
+
+// NewPoolTransactionsByPriceAndNonce creates a transaction set that can retrieve
+// price sorted transactions in a nonce-honouring way.
+//
+// Note, the input map is reowned so the caller should not interact any more with
+// if after providing it to the constructor.
+func NewPoolTransactionsByPriceAndNonce(signer Signer, txs map[common.Address]PoolTransactions) *PoolTransactionsByPriceAndNonce {
```
Comment (rlan35) [inline core/types/tx_pool.go:143]: no functional changes on these code, right?

File: core/state_processor.go [hunk 3]
```diff
@@ -79,41 +79,79 @@ func (p *StateProcessor) Process(
...
 	)
-	beneficiary, err := p.bc.GetECDSAFromCoinbase(header)
 
+	beneficiary, err := p.bc.GetECDSAFromCoinbase(header)
 	if err != nil {
 		return nil, nil, nil, 0, nil, err
 	}
 
-	// Iterate over and process the individual transactions
-	for i, tx := range block.Transactions() {
```
Comment (rlan35) [inline core/state_processor.go:89]: need a forking condition on pre-staking epoch, have to use the old logic before that.

Comment (gupadhyaya) [reply core/state_processor.go:89]: staking transactions won't be included in the block due to prestaking check in the `CommitTransactions` right?

File: core/types/tx_pool.go (hunk 2 above)
Comment (gupadhyaya) [reply core/types/tx_pool.go:143]: yes...

**Pull request merged**